src/main/java/com/softserveinc/dokazovi/service/impl/PostServiceImpl.java

```diff
@@ -1,23 +1,25 @@
 package com.softserveinc.dokazovi.service.impl;
 
-import com.softserveinc.dokazovi.entity.PostEntity;
+import com.softserveinc.dokazovi.dto.post.LatestPostDTO;
+import com.softserveinc.dokazovi.entity.enumerations.PostStatus;
+import com.softserveinc.dokazovi.mapper.PostMapper;
 import com.softserveinc.dokazovi.repositories.PostRepository;
 import com.softserveinc.dokazovi.service.PostService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class PostServiceImpl implements PostService {
 
 	private final PostRepository postRepository;
-
-	public PostServiceImpl(PostRepository postRepository) {
-		this.postRepository = postRepository;
-	}
+	private final PostMapper postMapper;
 
 	@Override
-	public Page<PostEntity> findAll(Pageable pageable) {
-		return postRepository.findAll(pageable);
+	public Page<LatestPostDTO> findAllByStatus(PostStatus postStatus, Pageable pageable) {
```

Comment:
Should we return posts other than with "PUBLISHED" status for "latest" in some cases?

Reply:
You're right. Fixed.

Reply:
Looks good!